src/Router/Entities/RouteParams.php

```diff
@@ -50,20 +50,17 @@ private function getParams(): array
 
         foreach ($actionParams as $actionParam) {
             /** @var string|null $paramType */
-            $paramType = null;
-
-            if ($actionParam->getType() && is_a($actionParam->getType(), ReflectionNamedType::class)) {
-                $paramType = $actionParam->getType()->getName();
-            }
+            $paramType = $actionParam->getType()?->__toString();
 
             $paramName = $actionParam->getName();
 
             $value = match ($paramType) {
-                'string' => $pathParams[$paramName] ?? '',
-                'int' => (int)($pathParams[$paramName] ?? 0),
-                'float' => (float)($pathParams[$paramName] ?? 0.0),
-                'bool' => (bool)json_decode($pathParams[$paramName] ?? '0'),
-                null => null,
+                'string' => $pathParams[$paramName],
+                'int' => (int)$pathParams[$paramName],
+                'float' => (float)$pathParams[$paramName],
+                'bool' => (bool)json_decode($pathParams[$paramName]),
+                null => throw UnsupportedParamTypeException::nonTyped(),
+                default => throw UnsupportedParamTypeException::fromType($paramType),
```

Comment:
@Chemaclass @JesusValera I need help there, as result of changes on commit https://github.com/gacela-project/router/pull/29/commits/1bbaa4872ae0d988d3038b7f9b63e8f9daa16470 phpstan is throwing this error `     Ignored error pattern #Match expression does not handle remaining value: string# was not matched in reported errors.` And I don't quite understand why, does default not capture the rest of the options?

Reply:
Remove lines 8,9 from phpstan.neon